subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -945,9 +960,21 @@ static int tls_mbedtls_handshake(struct tls_context *context, bool block)
 				ret = -ETIMEDOUT;
 				break;
 			}
+		} else {
+			/* MbedTLS API documentation requires session to
+			 * be reset in other error cases
+			 */
+			tmp = ret;
+			ret = tls_mbedtls_reset(context);
+			if (ret == 0) {
+				NET_ERR("TLS handshake error: -%x", -tmp);
```

Comment:
I think it'd make sense to log the error in any case (i. e. before the reset), it contains valuable information about the reason of the failure. No need for the extra variable then.

Reply:
I agree thank you!